iota/types.py

```diff
@@ -36,830 +35,829 @@
-    # for compatibility, we will continue to use ``choice`` in a loop.
-    # https://docs.python.org/3/library/random.html#random.choices
-    return cls(
-      ''.join(chr(generator.choice(alphabet)) for _ in range(length))
-        .encode('ascii')
-    )
-
-  @classmethod
-  def from_bytes(cls, bytes_, codec=AsciiTrytesCodec.name, *args, **kwargs):
-    # type: (Type[T], Union[binary_type, bytearray], Text, *Any, **Any) -> T
     """
-    Creates a TryteString from a sequence of bytes.
-
-    :param bytes_:
-      Source bytes.
-
-    :param codec:
-      Reserved for future use.
-
-      See https://github.com/iotaledger/iota.lib.py/issues/62 for more
-      information.
-
-    :param args:
-      Additional positional arguments to pass to the initializer.
-
-    :param kwargs:
-      Additional keyword arguments to pass to the initializer.
-    """
-    return cls(encode(bytes_, codec), *args, **kwargs)
-
-  @classmethod
-  def from_unicode(cls, string, *args, **kwargs):
-    # type: (Type[T], Text, *Any, **Any) -> T
-    """
-    Creates a TryteString from a Unicode string.
-
-    :param string:
-      Source string.
-
-    :param args:
-      Additional positional arguments to pass to the initializer.
+    A string representation of a sequence of trytes.
+
+    A tryte string is similar in concept to Python's byte string, except
+    it has a more limited alphabet.    Byte strings are limited to ASCII
+    (256 possible values), while the tryte string alphabet only has 27
+    characters (one for each possible tryte configuration).
+
+    IMPORTANT: A TryteString does not represent a numeric value!
+    """
+    @classmethod
+    def random(cls, length):
+        # type: (int) -> TryteString
+        """
+        Generates a random sequence of trytes.
+
+        :param length:
+            Number of trytes to generate.
+        """
+        alphabet = list(itervalues(AsciiTrytesCodec.alphabet))
+        generator = SystemRandom()
+
+        # :py:meth:`SystemRandom.choices` wasn't added until Python 3.6;
+        # for compatibility, we will continue to use ``choice`` in a loop.
+        # https://docs.python.org/3/library/random.html#random.choices
+        return cls(
+            ''.join(chr(generator.choice(alphabet)) for _ in range(length))
+            .encode('ascii')
+        )
 
-    :param kwargs:
-      Additional keyword arguments to pass to the initializer.
-    """
-    return cls.from_bytes(
-      bytes_  = string.encode('utf-8'),
-      codec   = AsciiTrytesCodec.name,
-      *args,
-      **kwargs
-    )
-
-  @classmethod
-  def from_string(cls, *args, **kwargs):
-    """
-    Deprecated; use :py:meth:`from_unicode` instead.
+    @classmethod
+    def from_bytes(cls,  # Type[T]
+                   bytes_,  # Union[binary_type, bytearray]
+                   codec=AsciiTrytesCodec.name,  # Text
+                   *args,  # *Any
+                   **kwargs  # **Any
+                   ):  # -> T
```

Comment:
Type hints should start with `# type: `, and the final one should take the form `# type: (...) -> T` (see [PEP-484](https://www.python.org/dev/peps/pep-0484/#suggested-syntax-for-python-2-7-and-straddling-code) for more info).